dart/utils/SkelParser.cpp

```diff
@@ -695,40 +692,41 @@ simulation::WorldPtr readWorld(
 
     if (hasElement(physicsElement, "collision_detector"))
     {
-      std::string strCD = getValueString(physicsElement, "collision_detector");
+      const auto cdType = getValueString(physicsElement, "collision_detector");
 
-      if (strCD == "fcl_mesh")
+      if (cdType == "fcl_mesh")
       {
-        collision_detector = collision::FCLCollisionDetector::create();
+        collision_detector = collision::CollisionDetector::Factory::create("fcl");
         auto cd = std::static_pointer_cast<collision::FCLCollisionDetector>(
               collision_detector);
         cd->setPrimitiveShapeType(collision::FCLCollisionDetector::MESH);
         cd->setContactPointComputationMethod(
               collision::FCLCollisionDetector::DART);
       }
-      else if (strCD == "fcl")
+      else if (cdType == "fcl")
       {
-        collision_detector = collision::FCLCollisionDetector::create();
+        collision_detector = collision::CollisionDetector::Factory::create("fcl");
         auto cd = std::static_pointer_cast<collision::FCLCollisionDetector>(
               collision_detector);
         cd->setPrimitiveShapeType(collision::FCLCollisionDetector::PRIMITIVE);
         cd->setContactPointComputationMethod(
               collision::FCLCollisionDetector::DART);
       }
-      else if (strCD == "dart")
-        collision_detector = collision::DARTCollisionDetector::create();
-#if HAVE_BULLET_COLLISION
-      else if (strCD == "bullet")
-        collision_detector = collision::BulletCollisionDetector::create();
-#endif
       else
-        dtwarn << "Unknown collision detector[" << strCD << "]. "
+      {
+        collision_detector = collision::CollisionDetector::Factory::create(cdType);
+      }
+
+      if (!collision_detector)
+      {
+        dtwarn << "Unknown collision detector[" << cdType << "]. "
                << "Default collision detector[fcl_mesh] will be loaded.\n";
+      }
     }
 
     if (!collision_detector)
     {
-      collision_detector = collision::FCLCollisionDetector::create();
+      collision_detector = collision::CollisionDetector::Factory::create("fcl");
       auto cd = std::static_pointer_cast<collision::FCLCollisionDetector>(
             collision_detector);
       cd->setPrimitiveShapeType(collision::FCLCollisionDetector::MESH);
```

Comment:
Can we put the code to create the `fcl_mesh` collision detector into a helper function? I am worried that the two versions of this code (i.e. the one in the `cdType == "fcl_mesh" block and in the default block) will become out of sync.